chihuahua/src/hcall/wasmi.rs

```diff
@@ -679,6 +939,21 @@ impl Chihuahua for WasmiHostProvisioningState {
         self.get_expected_shutdown_sources().clone()
     }
 
+    #[inline]
+    fn get_current_stream_source_count(&self) -> usize {
+        self.get_current_stream_source_count().clone()
+    }
+
+    #[inline]
+    fn get_expected_stream_sources(&self) -> Vec<u64> {
+        self.get_expected_stream_sources().clone()
```

Comment:
Generally, these "getter" methods should not clone, and should return borrowed data leaving the decision to allocate up to the calling code if it is needed.

Reply:
The old getters return a concrete Vec. I will change all the getters.

Reply:
Update: It should return concrete. The reference refers to a Vec inside a Mutex, which becomes invalid once the mutex is released.